### PR TITLE
Add NUGET_SHOW_STACK environment variable to enable showing the stack

### PR DESCRIPTION
**Pull request by joelverhagen**
https://github.com/NuGet/Home/issues/2482

One missing piece is documentation. I am planning on adding a FAQ under the **NuGet CommandLine** heading on the [NuGet FAQ](https://docs.nuget.org/consume/nuget-faq#nuget-commandline).

@emgarten @pranavkm

### CODE REVIEW

File: src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        /// the user. In prerelease or dogfooding scenarios, it is useful to have a non-verbose
+        /// logging level but, in the case of an unhandled exception, print the full exception for
+        /// bug reporting.
+        /// </summary>
+        /// <returns>
+        /// True if the exception stack should be displayed to the user. False, otherwise.
+        /// </returns>
+        public bool ShouldShowStack()
+        {
+            var rawShowStack = _reader.GetEnvironmentVariable("NUGET_SHOW_STACK");
```
Comment (pranavkm) [inline src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs]: You could potentially cache this in the ctor - the value wouldn't change once the process is running.

File: src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            bool showStack;
+
+            if (!bool.TryParse(rawShowStack, out showStack))
```
Comment (pranavkm) [inline src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs]: `var showStack = string.Equals(showStackString, "true", StringComparison.OrdinalIgnoreCase);`

File: src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            {
+                showStack = false;
+            }
+
+            return showStack;
+        }
+
+        private static Lazy<ExceptionLogger> _instance = new Lazy<ExceptionLogger>(
```
Comment (pranavkm) [inline src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs]: Shouldn't require this to be lazy.
`public static readonly ExceptionLogger Instance { get; } = new ExceptionLogger(new EnvironmentVariableWrapper());` should suffice

**Comment by pranavkm** [comment]
:shipit:  from me.

File: src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs (hunk 3 above)
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs]: Fixed.

File: src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs (hunk 2 above)
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs]: Fixed.

File: src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs (hunk 1 above)
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs]: Fixed.

**Comment by emgarten** [comment]
This will be great :smile: :rocket:

**Pull request merged**